addon/ng2/utilities/change.ts

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+import { readWithPromise, writeWithPromise } from './fs-promise';
+
+export interface Change {
+  /**
+   *  True on success, false otherwise.
+   */
+  apply(): Promise<void>;
+
+  // The file this change should be applied to. Some changes might not apply to
+  // a file (maybe the config).
+  path: string | null;
+
+  // The order this change should be applied. Normally the position inside the file.
+  // Changes are applied from the bottom of a file to the top.
+  order: number | null;
+
+  // The description of this change. This will be outputted in a dry or verbose run.
+  description: string;
+}
+
+
+/**
+ * Will add text to the source code.
+ */
+export class InsertChange implements Change {
+
+  constructor(
+      public path: string,
+      private pos: number,
+      private toAdd: string,
+      public order?: number,
```

Comment:
`order` is `pos` in this case. And description shouldn't be passed in, maybe just `'Inserting "abc" in file ${path}@pos.'`